src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -141,6 +145,17 @@ private X509Certificate UserCertSelectionCallbackWrapper(string targetHost, X509
             return _userCertificateSelectionCallback(this, targetHost, localCertificates, remoteCertificate, acceptableIssuers);
         }
 
+        private X509Certificate ServerCertSelectionCallbackWrapper(string targetHost)
+        {
+            return _userServerCertificateSelectionCallback(this, targetHost);
```

Comment:
When the callback is invoked how many of SslStream's properties will have been initialized? What can users inspect about the handshake?

Reply:
@Tratcher - not much, this is one of the first things being sent - what do you think might be useful at this stage? I think we might fix this past this PR if there is a use case.

Reply:
Post this PR I would like to see the available cipher suites as it's becoming more and more common to have both RSA and ECDSA certs using the more secure and smaller elliptical cert if the client can support it.

Reply:
Regardless of where we end up, the availability of specific fields in the callback needs to be clearly documented.